packages/errors/src/errors.ts

```diff
@@ -598,14 +609,29 @@ export const AllCypressErrors = {
 
         ${fmt.highlightSecondary(error)}`
   },
+  CLOUD_PROTOCOL_UPLOAD_STREAM_STALL_FAILURE: (error: Error & { chunkSizeKB: number, maxActivityDwellTime: number }) => {
+    const kbpsThreshold = (error.chunkSizeKB * 8) / (error.maxActivityDwellTime / 1000)
+
+    return errTemplate`\
+        Warning: We encountered slow network conditions while uploading the Test Replay recording for this spec.
+
+        The upload transfer rate fell below ${fmt.highlightSecondary(`${kbpsThreshold}kbps`)} over a sampling period of ${fmt.highlightSecondary(`${error.maxActivityDwellTime}ms`)}.
+
+        If this error occurs often, the sampling period may be configured by setting the ${fmt.highlightSecondary('CYPRESS_PROTOCOL_UPLOAD_SAMPLING_INTERVAL')} environment variable to a higher value than ${fmt.stringify(error.maxActivityDwellTime)}.
+
+        To prevent long CI executing durations, this Test Replay recording will not be uploaded.
```

Comment:
```suggestion
        To prevent long CI execution durations, this Test Replay recording will not be uploaded.
```

Reply:
@cacieprins This still needs addressing.

Reply:
Addressed in https://github.com/cypress-io/cypress/pull/30235/commits/397448d4084f5e040ac27632d7d1c22a6ececfef